cp23/hpi-fun2/motivation/notes.md

```diff
@@ -0,0 +1,66 @@
+# Motivation
+
+The problem is about finding the path coverage of the tree. 
+
+## Solution Idea
+
+> 1. Mark each node as reachable or unreachable (nodes, that exceed the allowed path length).
+>    - _The number of unreachable nodes is the second part of the solution_.
+>    - Also nodes, that have an unreachable parent are themselves unreachable, even if they do not exceed the path length.
+> 2. Find all leafs of the tree (can be done as part of step 1). 
+>    - A leaf is a node, that has no reachable children
+>    - Note, that this can also be a node, that is traditionally considered an inner node.
+> 3. Traverse the tree bottom-up:
+>    - The number of coaches a node needs is defined by `max(1, #(reachable_children) - 1)`
```

Comment:
This is the crucial point. Your solution do not have to be so extensive. You can keep it rather short.
You could write smth. like
![CodeCogsEqn(1)](https://github.com/compprog-lecture-tools/fun-contest-2023/assets/33495148/a1a9278a-ab41-4ba8-af89-dfa9f2ccb0cc)
and "then do tree dp". but you can also explain a bit more if you wish.

Reply:
Done 👍🏼